source/mechanics/fiber_network.cc

```diff
@@ -0,0 +1,62 @@
+#include <fiddle/mechanics/fiber_network.h>
+#include <deal.II/grid/tria.h>
+#include <deal.II/base/array_view.h>
+#include <deal.II/base/table.h>
+
+namespace fdl
+{
+    using namespace dealii;
+
+    template <int dim, int spacedim>
+    FiberNetwork<dim, spacedim>::FiberNetwork(const Triangulation<dim, spacedim> &tria,
+    const std::vector<std::vector<Tensor<1, spacedim>>> &fibers)
+    : tria(&tria)  
+    {
+        AssertThrow(fibers[0].size()>=0,
+                  dealii::ExcMessage("fiber vector is empty"));
+
+

```

Comment:
```suggestion
```
1. This will crash if `fibers` is empty (we should support that trivial case)
2. We already have `using namespace dealii;` so we don't need to namespace the exception.